src/V3Width.cpp

```diff
@@ -2150,14 +2155,40 @@ class WidthVisitor final : public VNVisitor {
         // Make sure dtype is sized
         nodep->dtypep(iterateEditMoveDTypep(nodep, nodep->subDTypep()));
         UASSERT_OBJ(nodep->dtypep(), nodep, "No dtype determined for var");
-        if (const AstUnsizedArrayDType* const unsizedp
-            = VN_CAST(nodep->dtypeSkipRefp(), UnsizedArrayDType)) {
-            if (!(m_ftaskp && m_ftaskp->dpiImport())) {
-                UINFO(9, "Unsized becomes dynamic array " << nodep << endl);
-                AstDynArrayDType* const newp
-                    = new AstDynArrayDType{unsizedp->fileline(), unsizedp->subDTypep()};
-                nodep->dtypep(newp);
-                v3Global.rootp()->typeTablep()->addTypesp(newp);
+        if (!(m_ftaskp && m_ftaskp->dpiImport())) {
+            AstNodeDType *dtp = nodep->dtypep();
+            AstNodeDType *np = nullptr;
+            while (VN_IS(dtp->skipRefp(), UnsizedArrayDType)
+                    || VN_IS(dtp->skipRefp(), UnpackArrayDType)
+                    || VN_IS(dtp->skipRefp(), AssocArrayDType)
+                    || VN_IS(dtp->skipRefp(), WildcardArrayDType)
+                    || VN_IS(dtp->skipRefp(), QueueDType)) {
+                if (const AstUnsizedArrayDType* const unsizedp
+                    = VN_CAST(dtp->skipRefp(), UnsizedArrayDType)) {
+                    UINFO(9, "Unsized becomes dynamic array ");
+                    if (!np) {
+                        // for Var itself
+                        UINFONL(9, nodep << endl);
+                    } else {
+                        // for subDType
+                        UINFONL(9, dtp << endl);
+                    }
```

Comment:
```suggestion
                    if (!np) {
                        UINFO(9, "Unsized becomes dynamic array (var itself) " << nodep);
                    } else {
                        UINFO(9, "Unsized becomes dynamic array (subDType) " << dtp);
                    }
```

Reply:
Probably better to keep `<< endl` at the end?

Reply:
Oops, yes ;)